crates/common/consensus/src/execution_engine/blob_versioned_hashes.rs

```diff
@@ -0,0 +1,33 @@
+use alloy_primitives::B256;
+use alloy_rlp::Decodable;
+use anyhow::anyhow;
+
+use super::{
+    new_payload_request::NewPayloadRequest,
+    rpc_types::transaction::{BlobTransaction, TransactionType},
+};
+use crate::deneb::execution_payload::ExecutionPayload;
+
+pub fn blob_versioned_hashes(execution_payload: &ExecutionPayload) -> anyhow::Result<Vec<B256>> {
+    let mut blob_versioned_hashes = vec![];
+    for transaction in execution_payload.transactions.iter() {
+        if TransactionType::try_from(&transaction[..])
+            .map_err(|err| anyhow!("Failed to detect transaction type: {err:?}"))?
+            == TransactionType::BlobTransaction
+        {
+            let blob_transaction = BlobTransaction::decode(&mut &transaction[1..])?;
+            blob_versioned_hashes.extend(blob_transaction.blob_versioned_hashes);
+        }
+    }
+
+    Ok(blob_versioned_hashes)
+}
+
+/// Return ``True`` if and only if the version hashes computed by the blob transactions of
+/// ``new_payload_request.execution_payload`` matches ``new_payload_request.versioned_hashes``.
+pub fn is_valid_versioned_hashes(new_payload_request: &NewPayloadRequest) -> anyhow::Result<bool> {
+    Ok(
+        blob_versioned_hashes(&new_payload_request.execution_payload)?
+            == new_payload_request.versioned_hashes,
+    )
+}
```

Comment:
can you move this function back to `crates/common/execution_engine` as it isn't used in this crate